src/core/tests/constant.cpp

```diff
@@ -1232,6 +1627,20 @@ TEST(constant, uint64_string_max) {
     }
 }
 
+//
+// nf4
+//
+TEST(constant, nf4_write_custom_type) {
+    Shape shape{3};
+    std::vector<TestDType> input{{-1.1f}, {-.5f}, {2.0f}};
```

Comment:
NF4 has a range [-1,1]. Even though this is a storage format, I think the right behavior is to through an exception for values out of the range, i.e. `-1.1f` and `2.0f`. This should help to avoid misuse.

Reply:
This PR doesn't change behavior of NF4 in Constant. The [NF4 tests](https://github.com/praasz/openvino/blob/master/src/core/tests/nf4.cpp#L34) or [python API test](https://github.com/praasz/openvino/blob/master/src/bindings/python/tests/test_transformations/test_compression_4bit.py#L19) assume that NF4 quantization will clamp value to the NF4 limits.

If this additional check is required lets make separate PR to see consequences and not block adding new types into Constant.